policy/modules/kernel/kernel.te

```diff
@@ -443,6 +443,7 @@ optional_policy(`
 ')
 
 optional_policy(`
+	libs_load_firmware(kernel_t)
```

Comment:
also `allow kernel_t self:system firmware_load;` might be needed.

Reply:
I am aware of such failure, however I don't know which firmware is causing it...

The system appears to be working fine without granting such permission and even more importantly firmware files are normally labelled as _lib_t_ rather than _kernel_t_.

Reply:
Hmm, it might have to be only needed for Linux 6.15; I removed this permission recently using 6.16.

Reply:
>> Hmm, it might have to be only needed for Linux 6.15; I removed this permission recently using 6.16.
>
> I am using kernel version 6.16.4 and the libs_load_firmware(kernel_t) permission is absolutely needed in order to properly load the firmware for all devices that rely on it.
>
> The kernel (running as kernel_t) needs to load firmware files from /lib/firmware which are binary files labelled as lib_t.

That comment was about `allow kernel_t self:system firmware_load;`, i think 6.15 did not correctly always took the target label from the firmware file.
I agree that `libs_load_firmware(kernel_t)` is necessary.